Added file on index_select (#5673)
Codecademy/docs: content/pytorch/concepts/tensor-operations/terms/index_select/index_select.md

---
Title: '.index_select()'

> Description: 'Extracts specific elements from a tensor to return a new tensor.'

Reviewer: ```suggestion
Description: 'Extract specific elements from a tensor and return a new tensor.'
```

Subjects:
  - 'AI'
  - 'Data Science'
Tags:
  - 'AI'
  - 'Arrays'
  - 'Data Structures'
  - 'Deep Learning'
CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'paths/computer-science'
---

> **Index_select()** is used to extract specific elements (rows, columns) from the tensor to return a new tensor. 

Reviewer: ```suggestion
**`Index_select()`** is used to extract specific elements (rows, columns) from the tensor to return a new tensor. 
```
The function name must be wrapped in backticks.

## Syntax

```pseudo

> torch.index_select(input,dim,index,out)

Reviewer: ```suggestion
torch.index_select(input, dim, index, out)
```
It would be helpful to add a note below regarding the syntax: 'If the result is to be stored in an existing tensor, the output tensor can be specified using the optional out argument.'

```

> - `input`: The input tensor
> - `dim`: The dimension to index (0 for rows, 1 for columns)
> - `index`: The 1-D tensor containing the indices to select
> - `out` (optional): The output tensor (optional)

Reviewer: ```suggestion
- `input`: The input tensor.
- `dim`: The dimension to index (0 for rows, 1 for columns).
- `index`: The 1-D tensor containing the indices to select.
- `out` (optional): The output tensor (optional).
```

## Example

> The following example demonstrates the usage of `.index_select()`:

Reviewer: ```suggestion
The following example demonstrates the use of `.index_select()`:
```

```py
import torch

#Define a tensor
ten = torch.tensor([[1, 2, 3, -8],
                    [4, 3, 8, 0],
                    [-1, 7, 6, 3],
                    [5, 6, 9, 0]])

indices = torch.tensor([0, 2])

#Define a new tensor using index_select 
r = torch.index_select(ten, 0, indices)
print(r)

#Define a new tensor using index_select 
c = torch.index_select(ten, 1, indices)
print(c)
```

```shell

> tensor([[1, 2, 3, -8],
>         [-1, 7, 6, 3]])
> tensor([[1, 4, -1, 5],
>         [3, 8, 6, 9]])

Reviewer: ```suggestion
tensor([[1, 2, 3, -8],
        [-1, 7, 6, 3]])
tensor([[1, 4, -1, 5],
        [3, 8, 6, 9]])
```
This output incorrectly selects values from rows instead of columns, which is incorrect, so please correct the way of printing this output.
